### ISSUE TITLE
Adjust clinical data validation

### ISSUE THREAD

**Issue by pieterlukasse** [cBioPortal/cbioportal#808]
Clinical data should now be provided in two files: 
- patient clinical data file
- samples clinical data file

The following parts need to be adjusted to comply to this new standard:
- [x] validation script (python)
- [x] unit tests of validation script
- [x] importer script (python)
- [x] ImportClinicalData.java 
- [x] TestImportClinicalData.java 
- [x] wiki documentation (at thehyve repo first: https://github.com/thehyve/cbioportal/wiki/File-Formats#clinical-data)
- [x] update wiki documentation at cbioportal repo https://github.com/cBioPortal/cbioportal/wiki/File-Formats#clinical-data

**Comment by pieterlukasse** [comment]
@zheins : please make a branch from our current dev branch: https://github.com/thehyve/cbioportal/tree/validator_and_loading_improvements

**Comment by ecerami** [comment]
@pieterlukasse:  can you explain a bit more why this is a good idea, e.g. what benefit do we get from doing this?

**Comment by pieterlukasse** [comment]
@ecerami : this is apparently the new way MSK has chosen to generate clinical data, as part of the latest release by @n1zea144 of the brca staging files which we are testing. Maybe @n1zea144 and @zheins can comment on why this changed? Personally I have no problem with this change as it makes the clinical file a bit less cumbersome (no need for the header specifying if an attribute is patient or sample attribute) and in theory it makes it possible to add new samples to your study without  the need for the patients file/data if the patient is already in the db.

**Comment by pieterlukasse** [comment]
another advantage of splitting up these files is that it is easier to validate. Once we have two files, we can actually introduce a constraint on patient data to ensure a patient is defined only once in a dataset. Now this is more difficult because all patient fields have to be repeated when adding a second sample for this patient, so the validation would have to compare all patient fields of this new record with the previous one to ensure nothing is being overwritten by mistake in the second record.

**Comment by n1zea144** [comment]
Even those the TCGA files are generated by the ETL, in general people that were manually curating a combined sample/patient file were finding it confusing, so we decided to split the files up.  For consistency, we decided to do this in the TCGA ETL pipeline too.  I don't think @zheins carried the changes into the scripts package, meaning the import class is still expecting the attribute type metadata.

**Comment by fedde-s** [comment]
I adjusted the validation script and the Python layer of the loading pipeline in thehyve/cbioportal@900ccb45bab815ee4eecbc218bbb85916543b41c. It is now out of synch with `ImportClinicalData.java` in its branch; the validator no longer recognises the header line that labels each column in a clinical attribute file as either patient-level or sample-level while the Java layer of the loading pipeline still requires it.

**Comment by fedde-s** [comment]
I believe @zheins agreed to adjust the Java layer accordingly, keeping the old deprecated format supported for internal needs expressed by @n1zea144. Reassigning.

**Comment by zheins** [comment]
I've made this change in RC. PR #937

**Comment by fedde-s** [comment]
Reassigning back to us at The Hyve to discuss how we will merge the wiki documentation.

**Comment by pieterlukasse** [comment]
documentation now in `rc`. Remaining issues in documentation (if any)  can be logged in a new ticket.
